### PR TITLE
Base Carousel Widget

### PR DESCRIPTION
**Pull request by AlexGStapleton**
This is the foundation of https://github.com/siteorigin/so-widgets-bundle/issues/1111

- Styling of navigation is left up to the widget. With that said, it's likely to be expanded in future commits/PR.

In hindsight, the Post Carousel Commit could have been spun off into a separate PR as it does exceed the scope of the title of the PR. Having it in this PR does however make it much easier to test. The Anything Carousel will definitely be in a separate PR.

### CODE REVIEW

**Comment by AlexGStapleton** [comment]
@Misplon Regarding the responsive settings, it makes sense to add a copy of all of the global widget settings. What are your thoughts on doing that? If you're keen, how do you want to handle the widget specific responsive settings? We could handle the fields like page builder and have an empty field mean the default value is used.

**Comment by Misplon** [comment]
Sounds good. If we could add in placeholder support though, that would be ideal vs blank fields when the default is being used.

**Comment by AlexGStapleton** [comment]
The last commit adds global responsive fields. It uses placeholders when the user is editing widgets to prevent the global settings from being saved.

`get_breakpoints()` is present to allow for continued support of the `siteorigin_widgets_post_carousel_breakpoints` filter. You can use the following code to confirm that filter still works as expected:

add_filter( 'siteorigin_widgets_post_carousel_breakpoints', function( $breakpoints ) {
	return array(
		'tablet_landscape' => 333,
		'tablet_portrait'  => 222,
		'mobile'           => 111,
	);
} );

A global `siteorigin_widgets_carousel_breakpoints` filter doesn't have a use case due to the responsive settings now being editable via widget settings. @Misplon If you're keen though, I can add that filter.

**Comment by Misplon** [comment]
Thanks, mate. We can add that filter at some point if we get a request.

**Review by Misplon — APPROVED** [review]

**Comment by Misplon** [comment]
@AlexGStapleton Can you take a look at the above conflict? Thanks, mate.

**Comment by AlexGStapleton** [comment]
@Misplon The last two commits could result in the Post Carousel styling not working as expected so re-testing will be required.

File: base/inc/widgets/base-carousel.class.php [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+					plugin_dir_url( SOW_BUNDLE_BASE_FILE ) . 'css/lib/slick.css',
+					array(),
+					'1.8.1'
+				),
+			)
+		);
+
+	}
+
+	// Allow widgets to override settings.
```
Comment (gregpriday) [inline base/inc/widgets/base-carousel.class.php:40]: Same here, just need to stick to using PHPDoc format.

**Review by gregpriday — CHANGES_REQUESTED** [review]

**Review by gregpriday — APPROVED** [review]

**Comment by Misplon** [comment]
@AlexGStapleton Just a small conflict here, thanks mate.

**Pull request merged**